### PR TITLE
Mediated transfers scenario

### PR DESCRIPTION
**Pull request by err508**

### CODE REVIEW

File: raiden/tests/scenarios/mediated-transfers-scnenarios.yaml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+          - assert: {from: 1, to: 0, total_deposit: 0,  balance: 0,  state: "opened"}
+          - assert: {from: 1, to: 2, total_deposit: 100, balance: 100, state: "opened"}
+          - assert: {from: 2, to: 1, total_deposit: 0,  balance: 0,  state: "opened"}
+          - assert: {from: 3, to: 2, total_deposit: 100,  balance: 100,  state: "opened"}
+          - assert: {from: 2, to: 3, total_deposit: 0, balance: 0, state: "opened"}
+
+      - serial:
+          tasks:
+          # In A-B-C with 100 tokens in each channel, send 1 token from A to C
+          - transfer: {from: 0, to: 2, amount: 1}
```
Comment (CosminNechifor) [inline raiden/tests/scenarios/mediated-transfers-scnenarios.yaml]: Can you please check for the ``expected_http_status: 200``?

File: raiden/tests/scenarios/mediated-transfers-scnenarios.yaml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+          # assert A - B - C
+          - assert: {from: 0, to: 1, total_deposit: 100, balance: 99, state: "opened"}
+          - assert: {from: 1, to: 0, total_deposit: 0,  balance: 1,  state: "opened"}
+          - assert: {from: 1, to: 2, total_deposit: 100, balance: 99, state: "opened"}
+          - assert: {from: 2, to: 1, total_deposit: 0,  balance: 1,  state: "opened"}
+      - wait: 1
+      - serial:
+          repeat: 10
+          tasks:
+          - transfer: {from: 0, to: 2, amount: 1}
```
Comment (CosminNechifor) [inline raiden/tests/scenarios/mediated-transfers-scnenarios.yaml]: Same as above, ``expected_http_status: 200``.

File: raiden/tests/scenarios/mediated-transfers-scnenarios.yaml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    first: 1
+    last: 4
+    template: "localhost:50{:02d}"
+  ## Alternatively give an explicit list:
+  #list:
+  #  - node1
+  #  - node2
+  #  - node3
+
+## Available task types:
```
Comment (CosminNechifor) [inline raiden/tests/scenarios/mediated-transfers-scnenarios.yaml]: Please remove the unnecessary comments.

File: raiden/tests/scenarios/mediated-transfers-scnenarios.yaml [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+          tasks:
+          # assert A - B -C
+          - assert: {from: 0, to: 1, total_deposit: 100, balance: 89, state: "opened"}
+          - assert: {from: 1, to: 0, total_deposit: 0,  balance: 11,  state: "opened"}
+          - assert: {from: 1, to: 2, total_deposit: 100, balance: 89, state: "opened"}
+          - assert: {from: 2, to: 1, total_deposit: 0,  balance: 11,  state: "opened"}
+      - serial:
+          repeat: 89
+          tasks:
+          - transfer: {from: 0, to: 2, amount: 1}
```
Comment (CosminNechifor) [inline raiden/tests/scenarios/mediated-transfers-scnenarios.yaml]: Same

File: raiden/tests/scenarios/mediated-transfers-scnenarios.yaml [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+          # assert A - B -C
+          - assert: {from: 0, to: 1, total_deposit: 100, balance: 0, state: "opened"}
+          - assert: {from: 1, to: 0, total_deposit: 0,  balance: 100,  state: "opened"}
+          - assert: {from: 1, to: 2, total_deposit: 100, balance: 0, state: "opened"}
+          - assert: {from: 2, to: 1, total_deposit: 0,  balance: 100,  state: "opened"}
+
+      - serial:
+          repeat: 100
+          tasks:
+          - transfer: {from: 3, to: 0, amount: 1}
```
Comment (CosminNechifor) [inline raiden/tests/scenarios/mediated-transfers-scnenarios.yaml]: same

File: raiden/tests/scenarios/mediated-transfers-scnenarios.yaml (hunk 1 above)
Comment (ulope) [reply raiden/tests/scenarios/mediated-transfers-scnenarios.yaml]: That is not necessary. The default is `2..` (i.e. any 2xx code is accepted).

**Comment by LefterisJP** [comment]
@err508 @CosminNechifor why has the review not been addressed yet. Seems to be pending 4 days now.

**Comment by CosminNechifor** [comment]
@err508 What about the cases I've mentioned in the above comment regarding the path finding and the one which ensure that we receive a 409 when the node is offline? Can you please add them? In rest it's fine.

**Comment by LefterisJP** [comment]
fixes https://github.com/raiden-network/raiden/issues/2562

**Pull request merged**